test/demo/raiseIntent/intentResolution.inttest.ts

```diff
@@ -0,0 +1,189 @@
+import 'jest';
+import 'reflect-metadata';
+
+import {Identity} from 'hadouken-js-adapter';
+import {_Window} from 'hadouken-js-adapter/out/types/src/api/window/window';
+
+import {Intent} from '../../../src/client/intents';
+import * as fdc3Remote from '../utils/fdc3RemoteExecution';
+import {TestAppData, setupTeardown, setupOpenDirectoryAppBookends, DirectoryTestAppData} from '../utils/common';
+import {testManagerIdentity, testAppInDirectory2, testAppUrl} from '../constants';
+import {IntentResolution, ResolveError, ResolveErrorMessage} from '../../../src/client/main';
+import {delay} from '../utils/delay';
+
+/**
+ * Intent registered by `testAppWithPreregisteredListeners1` right after opening
+ */
+const preregisteredIntent: Intent = {
+    type: 'test.IntentNamePreregistered',
+    context: {type: 'preregistered.IntentNamePreregisteredContext'}
+};
+const appHandlingIntent: DirectoryTestAppData = testAppInDirectory2;
+
+setupTeardown();
+
+type ContextListener = jest.Mock | (() => any);
+
+type TestParam = [
+    string,
+    ContextListener | undefined,
+    any // The value IntentResolution.data should resolve to
+];
+
+const resolutionTestParams: TestParam[] = [
+    ['a delayed value', async () => {
+        await delay(1000);
+        return 1;
+    }, 1],
+    ['a literal', () => 1, 1]
```

Comment:
Could these `1`s be something like `"intent-result"`?

Reply:
Changed to intent-result